app/controllers/groups_controller.rb

```diff
@@ -408,12 +408,13 @@ def remove_member
   def request_membership
     params.require(:reason)
 
-    unless current_user.staff?
-      RateLimiter.new(current_user, "request_group_membership", 1, 1.day).performed!
-    end
-
     group = find_group(:id)
-    group_name = group.name
+
+    begin
+      GroupRequest.create(group: group, user: current_user, reason: params[:reason])
+    rescue
```

Comment:
You should add a specific exception you want to catch here. Otherwise the error will occur on things like typos, nil exceptions, etc. Additionally, it should be `create!` if you want to raise an exception on validation error.